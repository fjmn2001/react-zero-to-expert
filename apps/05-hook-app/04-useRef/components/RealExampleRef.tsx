import MultipleCustomHooks from "../../03-examples/components/MultipleCustomHooks"

import "./../css/index.css"
import { useState } from "react"

const RealExampleRef = () => {
  const [show, setShow] = useState(false)

  return (
    <>
      <h1>Real Example Ref</h1>
      <hr />

      {show && <MultipleCustomHooks />}

      <button className={"btn btn-primary mt=5"} onClick={() => setShow(!show)}>
        Toggle
      </button>
    </>
  )
}

export default RealExampleRef
