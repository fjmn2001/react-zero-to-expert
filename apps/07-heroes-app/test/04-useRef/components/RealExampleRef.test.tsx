import { describe, expect, test } from "vitest"
import { render, screen, fireEvent } from "@testing-library/react"
import RealExampleRef from "../../../04-useRef/components/RealExampleRef"

describe("testing RealExampleRef.tsx", () => {
  test("it should show correctly", () => {
    const { container } = render(<RealExampleRef />)
    expect(container).toMatchSnapshot()
  })

  test("it should show <MultipleCustomHooks/> component", () => {
    // @ts-ignore
    ;(async () => {
      render(<RealExampleRef />)
      fireEvent.click(screen.getByText("Toggle"))

      // Wait for page to update with query text
      const component = await screen.findByText("Breaking Bad Quotes")
      expect(component).not.toBeNull()
    })()
  })
})
