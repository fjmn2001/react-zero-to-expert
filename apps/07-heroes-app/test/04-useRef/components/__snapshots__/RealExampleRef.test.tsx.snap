// Vitest Snapshot v1

exports[`testing RealExampleRef.tsx > it should show correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    class="btn btn-primary mt=5"
  >
    Toggle
  </button>
</div>
`;
